app/grandchallenge/algorithms/models.py

```diff
@@ -103,6 +103,11 @@ class Algorithm(UUIDModel, TitleSlugDescriptionModel, ViewContentMixin):
         blank=True,
         on_delete=models.SET_NULL,
     )
+    optional_hanging_protocols = models.ManyToManyField(
+        "hanging_protocols.HangingProtocol",
+        related_name="optional_hanging_protocols_of_algorithm",
```

Comment:
In this case for the reverse lookup you would have something like `HangingProtocol.objects.filter(optional_hanging_protocols_of_algorithm=algorithm)`. Thats a bit verbose. Maybe `optional_for_algorithm`?

Similar comment for the other 3 apps.

Reply:
fixed